src/slowmokit/methods/linear_model/logistic_regression/logistic_regression.cpp

```diff
@@ -0,0 +1,200 @@
+/**
+ * @file methods/linear_model/logistic_regression/logistic_regression.cpp
+ *
+ * Implementation of the Logistic Regression main program
+ */
+
+#include "logistic_regression.hpp"
+
+template<class T>
+std::vector<double> LogisticRegression<T>::softmax(std::vector<T> x)
+{ // Softmax function
+  std::vector<double> exps(x.size());
+  double sum = 0.0;
+  for (int i = 0; i < x.size(); i++)
+  { // Iterating over vector to compute sum of all exponential values in vector
+    exps[i] = exp(x[i]);
+    sum += exps[i];
+  }
+  std::vector<double> values(x.size());
+  for (int i = 0; i < x.size(); i++)
+  { // Now computing actual values after applying softmax
+    exps[i] /= double(sum);
+  }
+  return exps;
+};
+
+template<class T>
+double LogisticRegression<T>::crossEntropy(std::vector<int> y,
+                                           std::vector<T> qi)
+{ // Gives gradient descent -(d(z)/d(theta))
+  double l = 0.0;
+  for (int i = 0; i < y.size(); i++)
+  {
+    l += (y[i] * qi[i]);
```

Comment:
It should be `y[i] * log(q[i])` here. Read about logloss function used for logistic regression